chaos-monkey-spring-boot/src/main/java/de/codecentric/spring/boot/chaos/monkey/watcher/outgoing/ChaosMonkeyWebClientWatcher.java

```diff
@@ -0,0 +1,75 @@
+package de.codecentric.spring.boot.chaos.monkey.watcher.outgoing;
+
+import de.codecentric.spring.boot.chaos.monkey.component.ChaosMonkeyRequestScope;
+import de.codecentric.spring.boot.chaos.monkey.component.ChaosTarget;
+import de.codecentric.spring.boot.chaos.monkey.configuration.AssaultProperties;
+import de.codecentric.spring.boot.chaos.monkey.configuration.WatcherProperties;
+import java.util.Random;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.reactive.function.client.ClientRequest;
+import org.springframework.web.reactive.function.client.ClientResponse;
+import org.springframework.web.reactive.function.client.ExchangeFilterFunction;
+import org.springframework.web.reactive.function.client.ExchangeFunction;
+import reactor.core.publisher.Mono;
+
+/** @author Marcel Becker */
+public class ChaosMonkeyWebClientWatcher implements ExchangeFilterFunction {
+
+  private final ChaosMonkeyRequestScope chaosMonkeyRequestScope;
+  private final WatcherProperties watcherProperties;
+  private final AssaultProperties assaultProperties;
+
+  public ChaosMonkeyWebClientWatcher(
+      final ChaosMonkeyRequestScope chaosMonkeyRequestScope,
+      final WatcherProperties watcherProperties,
+      AssaultProperties assaultProperties) {
+    this.chaosMonkeyRequestScope = chaosMonkeyRequestScope;
+    this.watcherProperties = watcherProperties;
+    this.assaultProperties = assaultProperties;
+  }
+
+  @Override
+  public Mono<ClientResponse> filter(
+      ClientRequest clientRequest, ExchangeFunction exchangeFunction) {
+    Mono<ClientResponse> response;
+    response = exchangeFunction.exchange(clientRequest);
+    if (watcherProperties.isWebClient()) {
+      try {
+        chaosMonkeyRequestScope.callChaosMonkey(
+            ChaosTarget.WEB_CLIENT, this.getClass().getSimpleName());
```

Comment:
Change the `this.getClass().getSimpleName()` to the request's host:

Here is a snippet I'm using in my WebClient (similar to Micrometer approach for metrics for WebClient)
```
    private fun getClientName(request: ClientRequest): String {
        return request.url().host ?: return "unknown_host"
    }
```

Reply:
I agree, I will change it

Reply:
I wonder if we should change the `.callChaosMonkey()` method to take an Object instead of a String so the name mapper could leverage the objects passed in instead of parsing strings.